index/scorch/segment/zap/contentcoder.go

```diff
@@ -130,17 +147,15 @@ func (c *chunkedContentCoder) Add(docNum uint64, vals []byte) error {
 		c.currChunk = chunk
 	}
 
-	// mark the starting offset for this doc
-	dvOffset := c.chunkBuf.Len()
+	// mark the data size for this doc
 	dvSize, err := c.chunkBuf.Write(vals)
 	if err != nil {
 		return err
 	}
 
 	c.chunkMeta = append(c.chunkMeta, MetaData{
-		DocNum:   docNum,
-		DocDvLoc: uint64(dvOffset),
-		DocDvLen: uint64(dvSize),
+		DocNum:      docNum,
+		DocDvOffset: uint64(dvSize),
```

Comment:
Still find it a little disconcerting that a size is being stored in an offset.

One of the things I look for is to see if the "units" are the same.  Loc & Offset are synonyms for the same kind of units.  Len & Size are synonyms.  Kilograms & Weight are the same kind of thing.  At least in my head.

So, this has this weird feel to me of temporarily storing a kilogram-weight number into a telephone field.

Since the dvOffset is seemingly conveniently available (old/pink line number 134), wondering if storing that would be clearer, rather than remembering the length in the offset field?

Reply:
dvOffset+dvSize [loc type and size type]  are breaking the similar stuff analogy you mentioned here know? :) Something in that line could be possible. But I am not sure whether its entirely free from the flaw mentioned here?

Reply:
> dvOffset+dvSize [loc type and size type] are breaking the similar stuff analogy you mentioned here know? :) Something in that line could be possible. But I am not sure whether its entirely free from the flaw mentioned here?

oh, to my mind, an offset (or location) + a size (or a delta) leads to yet another offset.  Starting to think my mind is strange!  :-)

Reply:
Renaming "DocDvOffset" to "DocDvRunLen" might fix this , I think?